### ISSUE TITLE
Update target SDK version to 30

### ISSUE THREAD

**Issue by misaochan** [commons-app/apps-android-commons#4447]
Mandatory for app updates after Nov 2021. We should get started on it now ideally. If you want to submit a PR for this, please test all major app workflows on API 30 and report on your findings in your PR.

**Comment by animesh288** [comment]
Hey, Can I work on it?

**Comment by misaochan** [comment]
Sure, it is yours! :)

**Comment by animesh288** [comment]
How can I run the app in Beta or Prod version @misaochan 
Actually  I am new to open source , i have updated the SDK version but unable to run the app

**Comment by sivaraam** [comment]
@animesh288 You could run the app using an emulator or a real device. You could find some information in [this resource](https://developer.android.com/training/basics/firstapp/running-app) and [our wiki page](https://github.com/commons-app/commons-app-documentation/blob/master/android/Quick-start-guide-for-Developers.md).

@misaochan Are you sure a new-comer could handle this issue properly? I know it takes less effort someone who knows the app well or about Commons at least but a newcomer might not be familiar with the major workflows of the app and might find it difficult to ensure they tested it properly, no?

**Comment by animesh288** [comment]
Thanks @sivaraam but can you or @misaochan explain me the following part :

To be more specific , where to use this ":app:installBetaDebug"

![Screenshot 2021-06-10 235453](https://user-images.githubusercontent.com/68495725/121578002-ce720600-ca47-11eb-867d-7953fa3ff428.png)

**Comment by misaochan** [comment]
@sivaraam I reckon this is the best way to learn about the app's workflows. ;) If they give us a full report on what they tested, we can critique it, and if anything gets through it will likely be discovered on master (or at least pre-release testing) afterwards anyway. But I don't feel strongly about it, if you and @nicolas-raoul feel it should not be a "good first issue" we can remove it.

@animesh288 there should be a "Build variant" drop down menu in the lower left corner of your Android Studio. You can select "prodDebug" or "betaDebug" there. Using the terminal commands is optional.

**Comment by animesh288** [comment]
Yes I did that @misaochan but I was continuously getting this error.

![Screenshot 2021-06-11 144759](https://user-images.githubusercontent.com/68495725/121664523-331f7600-cac5-11eb-9d0a-0c5904b280b8.png)

**Comment by sivaraam** [comment]
@animesh288 That's weird. What version of Android Studio do you use?

Also, could share the commit hash of the revision of the repo you're using? You could get this using VCS -> Git -> Show current revision. A screenshot of the window that pops up would be useful.

**Comment by neslihanturan** [comment]
Hey @animesh288 can you try #4451 ? This might solve your build errors.

**Comment by animesh288** [comment]
@sivaraam  I am using studio 4.2.1
![Screenshot 2021-06-12 162449](https://user-images.githubusercontent.com/68495725/121773629-c1196080-cb9a-11eb-8837-a3b91a1afaa8.png)

**Comment by animesh288** [comment]
@neslihanturan Ok I'll try that. Thanks for help

**Comment by animesh288** [comment]
@neslihanturan 
I tried that 
It is still giving the same error

**Comment by sivaraam** [comment]
> @neslihanturan
> I tried that
> It is still giving the same error

When I try to build locally using that PR, I'm able to build successfully. It's interesting you're still getting the same error.

Anyways, could you let us know if you're running a usual 'Build' of the app or are you trying to run a specific task? Could you let us know how you're initiating the build?

**Comment by sivaraam** [comment]
Also, could you let us know the Gradle and JDK versions you're using?

**Comment by animesh288** [comment]
@sivaraam  Build successfully . There was some problem with the plugins and also some build errors as mentioned by @neslihanturan .
